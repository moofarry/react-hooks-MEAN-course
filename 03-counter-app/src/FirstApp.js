import React from "react";
import PropTypes from "prop-types";
// import React, { Fragment } from "react";

const FirstApp = ({ greeting,subtitle }) => {


  return (
    <>
      <h1> {greeting} </h1>
      {/* <pre>{JSON.stringify(greeting, null, 3)}</pre>  */}
      <p> {subtitle} </p>
    </>
  );
};

FirstApp.propTypes = {
  greeting: PropTypes.string.isRequired
}
FirstApp.defaultProps = {
  subtitle : 'Im sub-Title'
}

export default FirstApp;
