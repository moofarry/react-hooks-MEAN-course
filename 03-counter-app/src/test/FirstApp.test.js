import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import FirstApp from "../FirstApp";

describe("Test in <FirstApp />", () => {

  test("should show <FirstApp /> correctly", () => {
    const greeting = "Im Goku";
    const wrapper = shallow(<FirstApp greeting={greeting} />);

    expect(wrapper).toMatchSnapshot();
  });
  test("should show subtitle correctly", () => {
    const greeting = "Im Goku";
    const subTitle = 'Im sub-Title'
    const wrapper = 
      shallow(
        <FirstApp 
          greeting={greeting} 
        />
      );

    const textP = wrapper.find('p').text();
    expect(textP).toMatchSnapshot(subTitle);
  });
});
