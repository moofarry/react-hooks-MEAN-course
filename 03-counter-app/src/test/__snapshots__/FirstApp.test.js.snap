// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <FirstApp /> should show <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
     
    Im Goku
     
  </h1>
  <p>
     
    Im sub-Title
     
  </p>
</Fragment>
`;

exports[`Test in <FirstApp /> should show subtitle correctly: Im sub-Title 1`] = `" Im sub-Title "`;
